proton-j/src/main/java/org/apache/qpid/proton/engine/impl/CollectorImpl.java

```diff
@@ -57,9 +58,13 @@ public void pop()
         }
     }
 
-    public EventImpl put(Event.Type type, Object context)
+    public EventImpl put(EventType type, Object context)
     {
-        if (tail != null && tail.getType() == type &&
+        if (type == null)
+            throw new IllegalArgumentException("Type cannot be null");
+        if (!type.isValid())
+            throw new IllegalArgumentException("Cannot put events of type " + type);
+        if (tail != null && tail.getEventType() == type &&
```

Comment:
Can you please qualify 'the wrong thing', I really want to understand this?

With this overall change, I'm only working around the `final` of the enum, not around other aspects of it's semantics. This change brings us about on-par with proton-c imho where EventType === int.